### ISSUE TITLE
Implement logging infrastructure 

### ISSUE THREAD

**Issue by snizhana-dynnyk** [giantswarm/roadmap#311]
### User Story

- As a Giant Swarm engineer, I want to be able to access a history of logs for components managing our platform in order to be to both investigate ongoing operational issues and provide details for incident reports

### Tasks

- [x] ~https://github.com/giantswarm/giantswarm/issues/22335~
- [x] https://github.com/giantswarm/giantswarm/issues/22396
  - [x] https://github.com/giantswarm/roadmap/issues/722
  - [x] https://github.com/giantswarm/roadmap/issues/722
  - [x] https://github.com/giantswarm/giantswarm/issues/24855
  - [x] #2082
- [x] #2952
  - [x] https://github.com/giantswarm/giantswarm/issues/23432
  - [x] https://github.com/giantswarm/giantswarm/issues/26035
  - [x] https://github.com/giantswarm/roadmap/issues/2554
  - [x] #1839
  - [x] https://github.com/giantswarm/roadmap/issues/1849
  - [x] https://github.com/giantswarm/giantswarm/issues/24118
  - [x] https://github.com/giantswarm/roadmap/issues/2182
  - [x] #1903
  - [x] #2054
  - [x] https://github.com/giantswarm/roadmap/issues/2150
  - [x] https://github.com/giantswarm/giantswarm/issues/27146
  - [x] #2639
  - [x] https://github.com/giantswarm/giantswarm/issues/26964
  - [x] #1840 
- [x] #2756
  - [x] https://github.com/giantswarm/roadmap/issues/2761
  - [x] https://github.com/giantswarm/roadmap/issues/2758
  - [x] https://github.com/giantswarm/giantswarm/issues/28346
  - [x] #2834
  - [x] https://github.com/giantswarm/giantswarm/issues/28417
  - [x] https://github.com/giantswarm/giantswarm/issues/28416
  - [x] #2835
  - [x] https://github.com/giantswarm/giantswarm/issues/28426
- [x] #2951
  - [x] #2935
  - [x] #2942
  - [x] https://github.com/giantswarm/giantswarm/issues/29177
  - [x] #2943
  - [x] #2944
- [x] https://github.com/giantswarm/giantswarm/issues/28726

## Related
- https://github.com/giantswarm/giantswarm/issues/29073
- https://github.com/giantswarm/giantswarm/issues/25656
- https://github.com/giantswarm/giantswarm/issues/26706
- https://github.com/giantswarm/giantswarm/issues/28519
- https://github.com/giantswarm/giantswarm/issues/28813
- https://github.com/giantswarm/giantswarm/issues/27742
- https://github.com/giantswarm/giantswarm/issues/28879
- https://github.com/giantswarm/giantswarm/issues/29095
- #2771
- https://github.com/giantswarm/giantswarm/issues/27878
- https://github.com/giantswarm/giantswarm/issues/29259
- https://github.com/giantswarm/roadmap/issues/3178

**Comment by teemow** [comment]
The visibility is also interesting for customers. Similar to #182

**Comment by teemow** [comment]
via @anvddriesch: With CAPI a better interface towards distributed logging becomes more important. There are many controllers in the CAPI implementations compared to the single operator we use today. So in case you want to debug what is going on, eg during cluster creation, it would be good to have the logs in one place.

**Comment by snizhana-dynnyk** [comment]
as a requirement (coming from https://github.com/giantswarm/giantswarm/issues/11489) we want the k8s audit log kept

**Comment by JosephSalisbury** [comment]
![Uploading Screenshot 2022-03-15 at 13.59.24.png…]()

**Comment by JosephSalisbury** [comment]
<img width="1819" alt="Screenshot 2022-03-15 at 13 59 24" src="https://user-images.githubusercontent.com/297653/158394351-9cc7a69a-f627-4b64-aeac-c74b64d55c1b.png">

**Comment by TheoBrigitte** [comment]
#### Status update

- We do have a central Loki instance setup on a GiantSwarm cluster
- We are ingesting all logs (journald + pods)
- We are ingesting logs from some MCs (`anteater`, `gauss`, `gorilla`, `otter`)
- Cost is currently around 1$/month/MC

**Comment by TheoBrigitte** [comment]
Note from sig-product sync and further async discussions
https://docs.google.com/document/d/1Dbl_76JqeUlG16FhHU4P4kM41B3OP0HCkZLONXFH0f0/edit#
https://github.com/giantswarm/giantswarm/pull/24939

* We want to log as much as possible, but have alert in place in case some component log too much. So we can adjust log level and retention.
* Document current setup, retention policy.
* Loki as a managed app, for customer to monitor their app
* Centralized vs Installation setup
  * Some customer have security concern about data going out
  * Storage might be hard on installations
  * Automating S3 bucket creation  
* Is cost an issue ?

**Comment by TheoBrigitte** [comment]
The recent discussions around our logging architecture, about the centralized vs distributed approach pro and cons have been summarized into this [RFC](https://github.com/giantswarm/rfc/blob/main/logging-infrastructure/README.md).

We are no working on setting up Loki following the distributed approach.

**Comment by TheoBrigitte** [comment]
Added TODOs about Promtail deployment that were discussed in last refinement session.

**Comment by TheoBrigitte** [comment]
### Status update

We have a good progress on the dynamic configuration part, which we need in order to configure Grafana, Promtail, and the multitenant-proxy to make logs flow through our infrastructure. There are some last bit and pieces (more details here https://github.com/giantswarm/giantswarm/issues/27146) plus some testing to be done.
After this we enter testing phase, first on GiatnSwarm installations and then open the trial for customers.
So in short:
1. Test dynamic configuration
2. Testing on GiantSwarm installations
3. Trial on customer AWS vintage installations

**Comment by TheoBrigitte** [comment]
### Status update

* We successfully ran our pre-release on GiantSwarm AWS testing installations.
* We wrote a guide to evaluate Loki cost https://intranet.giantswarm.io/docs/observability/loki-cost-estimate/
* We are now ready to deploy Loki on AWS installations.
* We continue with implementing logging for AWS vintage WCs

**Comment by TheoBrigitte** [comment]
### Status update

* Logging infrastructure is live on all AWS installations
  * Since 28th of August on 19 installations
  * Since 20th of September on `bandicoot`, `flamingo`, and `fox`
* Documentation improved and made public in our handbook
  * Usage: https://handbook.giantswarm.io/docs/observability/loki-usage/
  * Cost estimate: https://handbook.giantswarm.io/docs/observability/loki-cost-estimate/
* AWS workload cluster system logs
  * Completed implementation
  * Running manual test

**Comment by QuentinBisson** [comment]
We are almost there, we are only missing https://github.com/giantswarm/giantswarm/issues/28726 so technically
